src/Google.Cloud.Functions.Hosting/HostingInternals.cs

```diff
@@ -171,6 +173,22 @@ internal static IServiceCollection AddServicesForFunctionTarget(IServiceCollecti
                     .AddSingleton<CloudEventFormatter>(new JsonEventFormatter())
                     .AddScoped(typeof(ICloudEventFunction), functionType);
             }
+            else if (GetGenericInterfaceImplementationTypeArguments(functionType, typeof(ITypedFunction<,>)) is Type[] signatureTypes)
+            {
+                Type requestType = signatureTypes[0];
+                Type responseType = signatureTypes[1];
+
+                services
+                    .AddScoped(typeof(IHttpFunction), typeof(TypedFunctionAdapter<,>).MakeGenericType(signatureTypes))
+                    .AddScoped(typeof(ITypedFunction<,>).MakeGenericType(signatureTypes), functionType);
+
+                // Provide JSON reader/writer implementations using System.Text.Json by default.
```

Comment:
thought: This seems a little random? Maybe consider adding a utility for generating the System.Text.Json reader/writer and find a way to annotate the typed function to specify those. It would be similar to the formatter for Cloud Events.

Reply:
Unlike the CloudEvents scenario, we don't have any common non-FF assembly that could contain an attribute specifying what reader/writer should be used.

But we're expecting JSON to be the default serialization format for most things. We may provide docs with an example of Json.NET, as that'll be reasonably common too.